src/jit/block.cpp

```diff
@@ -661,7 +661,7 @@ void BasicBlock::MakeLIR(GenTree* firstNode, GenTree* lastNode)
 bool BasicBlock::IsLIR()
 {
     const bool isLIR = (bbFlags & BBF_IS_LIR) != 0;
-    assert((bbTreeList == nullptr) || ((isLIR) == !bbTreeList->IsStatement()));
+    assert((isLIR && bbStmtList == nullptr) || (!isLIR && bbTreeList == nullptr));
```

Comment:
Not sure if I understood the condition before and not sure if I understand it now. Doesn't this make all empty blocks appear as LIR?

Reply:
(isLir == true) => ((bbStmtList == nullptr) && (bbTreeList == any)),
(isLir == false) => ((bbTreeList == nullptr) && (bbStmtList  == any)),
so it means lir should not have `bbStmtList `, hir should not have `bbTreeList `.